idseq_dag/util/s3.py

```diff
@@ -153,7 +153,8 @@ def refreshed_credentials(credentials_mutex=multiprocessing.RLock(), credentials
 
 def find_oldest_reference(refdir):
     try:
-        ls = subprocess.run(f"ls -t {refdir} | tail -1", shell=True, executable="/bin/bash", check=True, capture_output=True)
+        # To understand this ls command, please see the path forming explained in fetch_from_s3 when is_reference=True.
+        ls = subprocess.run(f"ls -td {refdir}/*/* | tail -1", shell=True, executable="/bin/bash", check=True, capture_output=True)
```

Comment:
We needed a '*/*' here after the design of fetch_from_s3 changed a while back to accommodate certain assumptions in certain steps (explained in comments in fetch_from_s3).   The result of not having the '*/*' is that the reference removal was too coarse (i.e. we would remove both NT and NR because they share the same folder in S3, even if one of them was touched) and it didn't take the correct timestamps into account (timestamp of cotnaining folder for NT and NR instead of the individual timetamp of NT and that of NR).

Reply:
This is now very thoroughly tested.   It really didn't matter much before adding the touching in this PR since any deletions were quite random anyway before that.

Reply:
Note that it changes the result of this function to a fully qualified path.